### ISSUE TITLE
No difference in status after clicking 'accept' or 'decline' buttons

### ISSUE THREAD

**Issue by vladimir-pankov** [dargmuesli/nuxt-cookie-control#121]
### Environment

browser

### Reproduction

watch(
  () => cookieControl.isConsentGiven.value,
  (current, _previous) => {
    console.log(`cookie isConsentGiven current=${JSON.stringify(current,null,4)}`)
    console.log(`cookie isConsentGiven previous=${JSON.stringify(_previous,null,4)}`)
},
  { deep: true }
)

logs the same:

cookie isConsentGiven current=true 
cookie isConsentGiven previous=false

for both buttons 'accept' and 'decline'

### Describe the bug

In both cases **isConsentGiven** changes from **false** to **true**:

const accept = () => {
  setCookies({
    **isConsentGiven: true,**
    cookiesOptionalEnabled: moduleOptions.cookies.optional,
  })
}

const decline = () => {
  setCookies({
    **isConsentGiven: true,**
    cookiesOptionalEnabled: moduleOptions.cookies.necessary,
  })
}

so no way to determine which choice is done.

Seems that setting 
**isConsentGiven: true,**
when decline() in not logically correct.

The changes 
**undefined --> true**  when accept()
**undefined --> false** when decline()
would look logically better.

### Additional context

If initial status of **isConsentGiven** is **undefined**
then it becomes possible to **watch** a change in

const declineAll = () => {
  setCookies({
    **isConsentGiven: false**,
    cookiesOptionalEnabled: [],
  })
}

which now is impossible.

**Comment by dargmuesli** [comment]
`isConsentGiven` applies to all cookies and `decline` only applies to optional cookies (could be named `declineOptional` instead). Declining optional cookies still consents to required cookies, so `isConsentGiven` is set to true.
`declineAll` applies to required cookies aswell, so no consent is given in that case.

To make it more clear I think it would make sense for me to change the naming a bit.

So in summary:
- use `isConsentGiven` to check if cookies were given consent **at all**
- use `cookiesEnabled` or `cookiesEnabledIds` so check if the cookie you're interested in was given consent

**Comment by vladimir-pankov** [comment]
"use isConsentGiven to check if cookies were given consent at all"
This is  ambiguous
isConsentGiven = true
means both:
1) optional+necessary cookies  were given consent ( by 'accept')
2) only necessary  cookies were given consent ( by 'decline' )

Yes, it is possible to distinguish by introduction of both
‘necessary' and ‘optional’  cookies but impossible without having ‘optional’ cookies. Documentation does not mention this feature.

**Comment by dargmuesli** [comment]
I don't really understand the answer. Is there anything you wish for? If so, please give a detailed specification :heart:

**Comment by dargmuesli** [comment]
Closing this as it seems reasonable to me to watch the enabled cookies. If you need button click events feel free to submit a PR! Should be fairly easy to implement :raised_hands:

**Comment by dennishelbig** [comment]
I also disagree with the current logic.
If you click decline, you expect that `isConsentGiven` will return `false`. 
Everything else is not intuitive.

Could you add the logic in the docs, so others can understand.
